### PR TITLE
Use available_providers by default

### PR DESCRIPTION
**Pull request by mgoin**
For ORT use all installed providers automatically

### CODE REVIEW

**Review by bfineran — COMMENTED** [review]
are there ort environment variables that can override this? also thinking about expected behavior when a user sets num_cores, but then will default to using ORT-GPU

**Review by InquestGeronimo — DISMISSED** [review]

**Comment by mgoin** [comment]
> are there ort environment variables that can override this? also thinking about expected behavior when a user sets num_cores, but then will default to using ORT-GPU

I don't know about environment variables for ORT, I think they don't use many? I think it would be reasonable to add a `providers` argument to the ORTEngine constructor so you could define it in code.

For num_cores, I think it still makes sense that will control the number of threads on the CPU while running on the GPU. I wouldn't be against disabling GPU with num_cores set

**Comment by bfineran** [comment]
> For num_cores, I think it still makes sense that will control the number of threads on the CPU while running on the GPU. I wouldn't be against disabling GPU with num_cores set

will leave this up to you - I think we should at least warn. Especially if a user has ORT-GPU installed but is attempting to benchmark deepsparse vs ort (cpu) head to head at the same num cores

File: src/deepsparse/benchmark/ort_engine.py [hunk 1]
```diff
@@ -98,11 +103,24 @@ def __init__(
...
+            if any(p != "CPUExecutionProvider" for p in providers):
+                _LOGGER.warn(
+                    (
+                        "Using ORTEngine with providers {} may not respect num_cores={},"
+                        " please specify CPUExecutionProvider"
+                    ).format(providers, num_cores)
+                )
+
         # TODO (michael): Unfortunately we are stacking overrides here, this can be
         # cleaned up once we pass the loaded ONNX around and not paths
```
Comment (rahul-tuli) [inline src/deepsparse/benchmark/ort_engine.py:125]: nitpick: Can ignore, could we remove this # TODO(s)?

**Review by rahul-tuli — COMMENTED** [review]
Looks good to me pending a few suggestions, on a side note I also have the following suggestions but github doesn't letme comment on those lines: 

ort_engine.py line 185: could we use `self.__class__.__qualname__` instead of `self.__class__.__name__`

More info on why is [here](https://stackoverflow.com/a/58110139)

File: src/deepsparse/benchmark/ort_engine.py [hunk 2]
```diff
@@ -98,11 +103,24 @@ def __init__(
...
+
         sess_options = onnxruntime.SessionOptions()
         sess_options.log_severity_level = 3
         if num_cores is not None:
             sess_options.intra_op_num_threads = num_cores
 
+            # Warn about num_cores usage for non-CPU providers
+            if any(p != "CPUExecutionProvider" for p in providers):
+                _LOGGER.warn(
+                    (
```
[the remark this answers is not in the document]
Comment (dbarbuzzi) [reply src/deepsparse/benchmark/ort_engine.py:118]: Ideally, we’d use neither and let the default formatting style of the logger handle the string interpolation: https://docs.python.org/3/howto/logging.html#optimization

[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/deepsparse/benchmark/ort_engine.py:118]: I agree with what's written there, but would like to note that this is not an expensive computation and would also maintain consistency with rest of the codebase.

**Review by rahul-tuli — APPROVED** [review]

**Review by InquestGeronimo — APPROVED** [review]

**Pull request merged**